File: app/src/main/java/de/bikebean/app/ui/drawer/history/battery/BatteryHistoryFragment.java
package de.bikebean.app.ui.drawer.history.battery;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;


import de.bikebean.app.R;
import de.bikebean.app.ui.drawer.history.HistoryAdapter;
import de.bikebean.app.ui.drawer.history.HistoryFragment;
import de.bikebean.app.ui.drawer.history.HistoryViewModel;

public class BatteryHistoryFragment extends HistoryFragment {

    @NonNull
    @Override
    public View onCreateView(final @NonNull LayoutInflater inflater,
                             final @Nullable ViewGroup container,
                             final @Nullable Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        final @NonNull View v = inflater.inflate(R.layout.fragment_history_battery, container, false);

        positionHistoryButton = v.findViewById(R.id.positionButton);
        batteryHistoryButton = v.findViewById(R.id.batteryButton);

        recyclerView = v.findViewById(R.id.recyclerView3);
        noDataText = v.findViewById(R.id.noDataText3);

        positionHistoryButton.setOnClickListener(super::navigateToTab);
        batteryHistoryButton.setOnClickListener(super::navigateToTab);

        return v;
    }

    @Override
    protected @NonNull HistoryViewModel getNewStateViewModel() {
        return new ViewModelProvider(this).get(BatteryHistoryViewModel.class);
    }

    @Override
    protected void setupListeners() {
        ((BatteryHistoryViewModel) st)
                .getBatteryConfirmed().observe(this, this::setStatesToAdapter);
    }

    @Override
    protected @NonNull HistoryAdapter getNewAdapter(final @NonNull Context ctx) {
        return new BatteryHistoryAdapter(ctx,
                ((BatteryHistoryViewModel) st).getBatteryConfirmed().getValue());
    }
}

File: app/src/main/java/de/bikebean/app/ui/drawer/status/battery/history/BatteryHistoryFragment.kt
package de.bikebean.app.ui.drawer.status.battery.history

import android.content.Context
import de.bikebean.app.ui.drawer.status.history.HistoryFragment
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import de.bikebean.app.R
import de.bikebean.app.ui.drawer.status.history.HistoryViewModel
import androidx.lifecycle.ViewModelProvider
import de.bikebean.app.ui.drawer.status.history.HistoryAdapter

class BatteryHistoryFragment : HistoryFragment() {

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View = inflater
            .inflate(R.layout.fragment_history_battery, container, false)
            .apply {
                recyclerView = findViewById(R.id.recyclerView3)
                noDataText = findViewById(R.id.noDataText3)
            }.also {
                super.onCreateView(inflater, container, savedInstanceState)
            }

    override val newStateViewModel: HistoryViewModel get() =
            ViewModelProvider(this).get(BatteryHistoryViewModel::class.java)

    override fun setupListeners() {
        (st as BatteryHistoryViewModel).batteryConfirmed.observe(this, ::setStatesToAdapter)
    }

    override fun getNewAdapter(ctx: Context): HistoryAdapter = BatteryHistoryAdapter(ctx,
            (st as BatteryHistoryViewModel).batteryConfirmed.value)

}